cmd/openvdc/cmd/copy/copy.go

```
@@ -0,0 +1,84 @@
+package copy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+
+	"github.com/axsh/openvdc/api"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+)
+
+type Client struct {
+	ClientConfig *ssh.ClientConfig
+	Host         string
+	Port         string
+}
+
+func NewClient(cr *api.CopyReply) (*Client, error) {
+	host, port, err := net.SplitHostPort(cr.GetAddress())
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("Invalid ssh host address: %s", cr.GetAddress()))
+	}
+
+	config := &ssh.ClientConfig{
+		User: cr.GetInstanceId(),
+	}
+
+	return &Client{
+		ClientConfig: config,
+		Host:         host,
+		Port:         port,
+	}, nil
+}
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	} else {
+		return true
+	}
+}
+
+func (c *Client) CopyFile(filePath string, instanceDir string) error {
+	if !fileExists(filePath) {
+		return errors.New("Source file not found")
+	}
+
+	file, _ := os.Open(filePath)
```

Please do not squash the error from IO operation.